### PR TITLE
Upgrade alluxio to 2.9.0

### PR DESCRIPTION
**Pull request by ssz1997**
Upgrade alluxio to 2.9.0

Solves https://github.com/fluid-cloudnative/fluid/issues/2332

### CODE REVIEW

File: charts/fluid/fluid/values.yaml [hunk 1]
```diff
@@ -41,11 +41,9 @@ runtime:
     controller:
       image: fluidcloudnative/alluxioruntime-controller:v0.9.0-5fe0498
     runtime:
-      # image: fluidcloudnative/alluxio:release-2.7.3-SNAPSHOT-a7154f1
```
Comment (ssz1997) [inline charts/fluid/fluid/values.yaml:44]: I think we don't need this comment anymore?

File: tools/alluxio/tarball.sh [hunk 2]
```diff
@@ -1,12 +1,12 @@
 set -x -e
 
 # download go package if not exist
-[ -e "go1.13.6.linux-amd64.tar.gz" ] \
-  || wget https://studygolang.com/dl/golang/go1.13.6.linux-amd64.tar.gz
+[ -e "go1.18.1.linux-amd64.tar.gz" ] \
```
Comment (ssz1997) [inline tools/alluxio/tarball.sh:4]: We upgraded the tarball generation to use go1.18.1: https://github.com/Alluxio/alluxio/pull/15881

**Comment by cheyang** [comment]
@ssz1997 

I'm trying to build the docker images with  the following command.  Could you help check what's wrong?

```
bash build-image.sh
  adding underfs module ufs-hadoop-3.2 to lib/ ... done
  adding underfs module ufs-hadoop-ozone-1.2.1 to lib/ ... done
  creating the new distribution tarball ... done
  removing the temporary repositories ... done
GIT_COMMIT=5a63178
Sending build context to Docker daemon  1.428GB
Step 1/10 : FROM alluxio/alluxio:2.9.1-SNAPSHOT
manifest for alluxio/alluxio:2.9.1-SNAPSHOT not found: manifest unknow
```

**Comment by ssz1997** [comment]
@cheyang 

The dev image now is based on the prod image and the image name in `FROM` is hard-coded, which means we have to build the prod image first and name it exactly as the hard-coded image name. 

I can make the hard-coded image name a build argument but most likely we won't cherry-pick it to 2.9.0. Plus we still have to build the base image first, so we still need to do a version comparison I think?

**Comment by ssz1997** [comment]
@cheyang Do we want to revert all the previous changes?

**Comment by cheyang** [comment]
> @cheyang Do we want to revert all the previous changes?

Yes, I suggest reverting the script change.

**Review by cheyang — APPROVED** [review]

**Pull request merged**